tiny_dnn/activations/softmax_layer.h

```diff
@@ -35,8 +35,15 @@ class softmax_layer : public activation_layer {
                            const vec_t &dy) override {
     const serial_size_t len = static_cast<serial_size_t>(dy.size());
 
-    // auxilliary vector to store element wise softmax gradients of all elements
-    vec_t df(len, 0);
+// auxilliary vector to store element wise softmax gradients of all elements
+
+#if HAS_CXX11_THREAD_LOCAL
+#define ATTRIBUTE_TLS thread_local
+#else
+#define ATTRIBUTE_TLS
+#endif
+
+    ATTRIBUTE_TLS vec_t df(len, 0);
```

Comment:
Since this is library is header only, it's a bit difficult to isolate preprocessor simbols defined in library header files.
So, using `#undef ATTRIBUTE_TLS` or simply writing in this can be preferrable.

```cpp
#if HAS_CXX11_THREAD_LOCAL
    thread_local
#endif
    vec_t df(len, 0);
```

Or putting preprocessor macros in `util/macro.h` would be good. But this `thread_local` matter is a bit complicated as discussed earlier. Why Apple didn't release C++11 conformant compiler sooner!